pkg/cli/render_csv.go

```diff
@@ -259,26 +268,99 @@ func (p *Project) parseToCSV(project *reports.Project, data *csvData) {
 			pSrv := project.Services[srv]
 			pSrvRes := project.Services[srv].Resources[res]
 
+			unit, val := humanReadable(p.Opts.HumanReadable, pSrvRes.ResourceInfo.Unit, rawValues{
+				"quota": pSrvRes.Quota,
+				"usage": pSrvRes.Usage,
+			})
+
 			switch {
 			case p.Opts.Names:
 				csvRecord = append(csvRecord, p.DomainName, project.Name, pSrv.ServiceInfo.Type,
-					pSrvRes.ResourceInfo.Name, strconv.FormatUint(pSrvRes.Quota, 10),
-					strconv.FormatUint(pSrvRes.Usage, 10), string(pSrvRes.ResourceInfo.Unit),
+					pSrvRes.ResourceInfo.Name, val["quota"], val["usage"], unit,
 				)
 			case p.Opts.Long:
 				csvRecord = append(csvRecord, p.DomainID, p.DomainName, project.UUID, project.Name, pSrv.ServiceInfo.Area,
 					pSrv.ServiceInfo.Type, pSrvRes.ResourceInfo.Category, pSrvRes.ResourceInfo.Name,
-					strconv.FormatUint(pSrvRes.Quota, 10), strconv.FormatUint(pSrvRes.Usage, 10),
-					string(pSrvRes.ResourceInfo.Unit), time.Unix(pSrv.ScrapedAt, 0).UTC().Format(time.RFC3339),
+					val["quota"], val["usage"], unit, time.Unix(pSrv.ScrapedAt, 0).UTC().Format(time.RFC3339),
 				)
 			default:
 				csvRecord = append(csvRecord, p.DomainID, project.UUID, pSrv.ServiceInfo.Type,
-					pSrvRes.ResourceInfo.Name, strconv.FormatUint(pSrvRes.Quota, 10),
-					strconv.FormatUint(pSrvRes.Usage, 10), string(pSrvRes.ResourceInfo.Unit),
+					pSrvRes.ResourceInfo.Name, val["quota"], val["usage"], unit,
 				)
 			}
 
 			*data = append(*data, csvRecord)
 		}
 	}
 }
+
+type rawValues map[string]uint64
+type convertedValues map[string]string
+
+// unitExp is map of limes.Unit to x, such that base-2 exponential of x
+// is the number of bytes for some specific unit.
+var unitExp = map[limes.Unit]float64{
+	limes.UnitKibibytes: 10,
+	limes.UnitMebibytes: 20,
+	limes.UnitGibibytes: 30,
+	limes.UnitTebibytes: 40,
+	limes.UnitPebibytes: 50,
+	limes.UnitExbibytes: 60,
+}
+
+func humanReadable(convert bool, unit limes.Unit, rv rawValues) (string, convertedValues) {
+	cv := make(convertedValues, len(rv))
+
+	// 'usage' is used to determine conversion suitability because it is common
+	// in all three hierarchies and is the lowest value amongst its counterparts
+	if unit == limes.UnitNone || lenUint(rv["usage"]) < 4 {
+		convert = false
+	}
+	if !convert {
+		for i, v := range rv {
```

Comment:
`i` sounds like an integer even though the map has string keys. Please prefer. `for key, value := range rv`. (Also in the other similar loops.)